### PR TITLE
Add optional setting to include tip in donation amount

### PR DESCRIPTION
**Pull request by davidcol25**
This feature will allow an organization to add the tip in the total donation amount. This feature is disabled by default and can be enabled in the charity settings.

### CODE REVIEW

**Review by kevinhughes27 — CHANGES_REQUESTED** [review]
Looks great! Nice stuff :smile_cat: 

I'd like to change the naming from `add_tip` to `include_tip` and in the user facing copy as well because I think that is a better way to express it to the user interface and in our code.

Another screenshot would be great here too. Just of the charity settings form.

I'd also like to see a new test added that uses an order fixture with a Tip (or modifies an existing fixture in a reasonable way) and the expected outcome

File: db/migrate/20220401184643_add_tip.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddTip < ActiveRecord::Migration[5.2]
+  def change
+      add_column :charities, :add_tip, :boolean, default: false, null: false
+  end
+end
```
Comment (kevinhughes27) [inline db/migrate/20220401184643_add_tip.rb:5]: nitpick but this file is missing a final new line.

Comment (kevinhughes27) [reply db/migrate/20220401184643_add_tip.rb:5]: I recommend finding the setting to in your editor to do this automatically

**Comment by nosdf227** [comment]
Settings

This is the screenshot for the Settings tab. You can see that we added the checkbox at the bottom.

**Comment by nosdf227** [comment]
Donations

Details for the `19.5` in the `Amount` column:
* Bought 2 items at $15 each (for a subtotal of $30)
* Percentage eligible for tax receipt : 50.0%
* Tip of $4.50

**Comment by davidcol25** [comment]
> I'd also like to see a new test added that uses an order fixture with a Tip (or modifies an existing fixture in a reasonable way) and the expected outcome

Could you maybe give us some guidance on how to create a new order fixture, can it be generated? Thank you.

**Comment by kevinhughes27** [comment]
The fixtures for these tests are just json files in [this directory](https://github.com/kevinhughes27/shopify-tax-receipts/blob/master/test/fixtures/order.json) that we load up and use in the test. This json blob is the same blob Shopify sends in the order webhook. You can also get the json for any order by adding `.json` to the url in your test shop admin. 

I'd recommend grabbing the json from the order you used for the screenshot above. You might need to edit the shop name to match the test shop which you can find in the other fixtures.

File: test/fixtures/order_with_tip.json [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+        }
+      },
+      "requested_fulfillment_service_id": null,
+      "source": "shopify",
+      "title": "Standard",
+      "tax_lines": [],
+      "discount_allocations": []
+    }
+  ]
+}
```
Comment (kevinhughes27) [inline test/fixtures/order_with_tip.json:411]: another nit on a missing newline here

**Review by kevinhughes27 — APPROVED** [review]
test looks good to me. Thanks for adding.

I just realized that I had CI slightly mis-configured and it is not running tests for forked branches. I have fixed the setting so if you push a blank commit or amend the last commit to change it then tests will run and we can see CI pass before I merge.

as part of this I also realized that your last PR https://github.com/kevinhughes27/shopify-tax-receipts/pull/219 broke some tests because the default order in the test now needs to reach out to Shopify for more information in order to render the home page. This can be fixed by adding more data to the donation in the test to prevent the api call or by mocking the api call. Can I get you to either put up a new PR for that fix or fix it in this PR? I'd like to get CI green before merging.

If you can get those things done I'll be able to merge this week and get the feature live! We could also make an announcement to users of the app about this because it has been requested a few times.

**Review by kevinhughes27 — APPROVED** [review]
I'll look into getting this deployed as soon as I can!

**Comment by davidcol25** [comment]
> I'll look into getting this deployed as soon as I can!

Alright, thank you so much!

**Comment by kevinhughes27** [comment]
Finally got this deployed! sorry for the delay. Had a couple things come up on my end.

**Pull request merged**